### DISCUSSION TITLE
Edits Class : Response body parse failed. See previous exception for details

### DISCUSSION THREAD

**Discussion by NeelDroid** [tectalichq/public-openai-client-php#11] [General] [answered]
As I can see that the response was generated but for some reason the CreateResponse : id is not returned

Take a look at this error trace -
![Screenshot_20230206_123943](https://user-images.githubusercontent.com/5361839/216908282-23c33594-4607-4dd4-b25d-7a32a6d85e2d.png)

**Comment by thejamescollins** [comment]
Thanks for letting us know @NeelDroid.

At first glance it looks like the `id` property is no longer being sent back by OpenAI's API.

We'll be looking into this soon.

In the interim, we would recommend using `toArray()` rather than `toModel()` for the `Edits::create()` endpoint response. That way you'll get back an associative array without any DTO validation.

https://tectalic.com/apis/openai/docs/php#retrieving-the-response

**Comment by thejamescollins** [comment]
Hi again @NeelDroid,

We have [reported this issue to the OpenAI team](https://github.com/openai/openai-openapi/pull/16), as their official API specification defines the `id` and `model` properties as being required for the Create Edits endpoint.

I'm unsure whether these properties have been removed from the API recently.

We will wait to hear back from them.

Thanks for your patience,

James

**Comment by thejamescollins** [comment, marked as answer]
Hi there @NeelDroid,

We have released version 1.3.1 today, and this new version contains a fix for this issue. Please see https://github.com/tectalichq/public-openai-client-php/releases/tag/v1.3.1 for details.

After you update to this new version, you should be able to retrieve a Model response for the `Edits::create()` handler method.
